addons/web_editor/static/src/js/editor/snippets/s_social_media/options.js

```diff
@@ -176,90 +152,91 @@ export const SocialMedia = options.registry.SocialMedia = options.Class.extend({
-            media = media.split('social_').pop();
-            if (!this.$target[0].querySelector(`:scope > a[href="/website/social/${media}"]`)) {
-                const entryNotInDom = this.entriesNotInDom.find(entry => entry.media === media);
-                if (!entryNotInDom) {
-                    this.entriesNotInDom.push({
-                        id: generateHTMLId(),
-                        display_name: link,
-                        placeholder: `https://${media}.com/yourPage`,
-                        undeletable: true,
-                        selected: false,
-                        listPosition: listPosition++,
-                        media: media,
-                        notToggleable: false,
-                    });
-                } else {
-                    // Do not change the listPosition of the existing entry.
-                    entryNotInDom.display_name = link;
-                    entryNotInDom.undeletable = true;
-                    entryNotInDom.notToggleable = false;
-                }
+        for (const [mediaName, mediaUrl] of await Object.entries(medias)) {
+            const entry = this._lastEntries.find(x => x.media === mediaName);
+            if (!entry) {
+                this._lastEntries.push({
+                    id: generateHTMLId(),
+                    display_name: mediaUrl,
+                    placeholder: `https://${mediaName}.com/yourPage`,
+                    undeletable: true,
+                    notToggleable: false,
+                    selected: false,
+                    media: mediaName,
+                });
             }
         }
-        // Reorder entries and entriesNotInDom by position.
-        entries = entries.concat(this.entriesNotInDom);
-        entries.sort((a, b) => {
-            return a.listPosition - b.listPosition;
-        });
-        return JSON.stringify(entries);
+
+        return JSON.stringify(this._lastEntries);
     },
     /**
-     * Fetches the urls of the social networks that are in the database.
+     * Finds the social network for the given entry.
+     *
+     * @param {String} entry
+     * @return {String | undefined} The social network to which the url leads to.
      */
-    async _fetchSocialMedia() {
-        if (!dbSocialValuesProm) {
-            let websiteId;
-            this.trigger_up('context_get', {
-                callback: function (ctx) {
-                    websiteId = ctx['website_id'];
-                },
-            });
-            // Fetch URLs for db links.
-            dbSocialValuesProm = this._rpc({
-                model: 'website',
-                method: 'read',
-                args: [websiteId, ['social_facebook', 'social_twitter', 'social_linkedin',
-                    'social_youtube', 'social_instagram', 'social_github']],
-            }).then(function (values) {
-                [dbSocialValues] = values;
-                delete dbSocialValues.id;
-            });
+    _getEntryMediaIcon(entry) {
```

Comment:
This does not work anymore (try typing `https://www.twitch.tv` as custom url for example).